backup/manager.go

```diff
@@ -272,25 +333,30 @@ func (b *Manager) Start() error {
 				}
 				var err error
 				var accountName string
+				var nodeID string
 
 				if req.BackupNodeData {
 					b.log.Infof("starting backup node data")
-					if accountName, err = b.processBackupRequest(true); err != nil {
+					if accountName, nodeID, err = b.processBackupRequest(true); err != nil {
 						b.log.Errorf("failed to process backup request: %v", err)
 						b.notifyBackupFailed(err)
 						continue
 					}
 				}
 				if req.BackupAppData {
 					b.log.Infof("starting backup app data")
-					if accountName, err = b.processBackupRequest(false); err != nil {
+					if accountName, nodeID, err = b.processBackupRequest(false); err != nil {
 						b.log.Errorf("failed to process backup request: %v", err)
 						b.notifyBackupFailed(err)
 						continue
 					}
 				}
 
-				b.db.markBackupRequestCompleted(pendingID)
+				err = b.db.markBackupRequestCompleted(pendingID)
+				if err != nil {
+					b.log.Errorf("markBackupRequestCompleted returned with %v", err)
+				}
+				b.updateLatestBackupTime(nodeID)
```

Comment:
This looks problematic performance wise. For every backup it will fetch all snapshots. Can we get the numbers of how much it takes for this call relative the the whole backup process?

Reply:
After a recommendation from Akiba and you we instead use a local json file which is a copy of the latest uploaded snapshot. This file is not backed up to cloud, but is only checked locally. This way we avoid having an old timestamp saved in the db when restoring on a new device. 


The performance when updating the json file is the following. This is also the time it takes to call the `AvailableSnapshots`, which i guess is the biggest bottleneck.

> 2023-05-04 08:51:41.987 [INF] BCKP: updateLatestBackupTime completed in 599.812458ms
2023-05-04 08:51:54.215 [INF] BCKP: updateLatestBackupTime completed in 481.7955ms
2023-05-04 08:52:02.039 [INF] BCKP: updateLatestBackupTime completed in 305.548834ms
2023-05-04 08:52:06.271 [INF] BCKP: updateLatestBackupTime completed in 386.940083ms
2023-05-04 08:52:20.032 [INF] BCKP: updateLatestBackupTime completed in 519.703375ms
2023-05-04 08:52:47.207 [INF] BCKP: updateLatestBackupTime completed in 412.387209ms
2023-05-04 08:53:16.362 [INF] BCKP: updateLatestBackupTime completed in 549.454875ms
2023-05-04 08:53:17.747 [INF] BCKP: updateLatestBackupTime completed in 682.496208ms
2023-05-04 08:53:32.654 [INF] BCKP: updateLatestBackupTime completed in 438.52775ms

Reply:
Worth checking if making the UploadBackupFiles reurn the latest snapshot (or the latest modified time) enable us not to query the available snapshot and by that reducing this time.